### PR TITLE
Remove netlify plugin

### LINKED ISSUE #1468: Plugin Netlify expects a datadir component in deploy func, causing a crash
**Describe the bug**

The Netlify platform still expects a `datadir.Component` to be passed through to its Deploy func. Because of this, it ends up with an error on the deploy step because of the missing expected component not being passed through. No other plugins use this, and it should be removed as an expected param to the Netlify plugin.

**Steps to Reproduce**

Thanks @jgwhite for the reproduction information

Attempt to deploy using the Netlify plugin

```hcl
# waypoint.hcl
project = "wp-netlify"
app "main" {
  build {
    use "files" {}
  }
  deploy {
    use "netlify" {
      site_id      = "<site id>"
      site_name    = "<site name>"
      access_token = "[REDACTED]"
    }
  }
}
```

Output

```
Deploying...
1 error occurred:
	* argument cannot be satisfied: type *anypb.Any (subtype: "hashicorp.waypoint.sdk.Args.DataDir.Component"
```

Debug Log

https://gist.github.com/jgwhite/8b4b2c2e0dec0d69341299743c51ec75

**Expected behavior**

Be able to deploy

**Waypoint Platform Versions**
Additional version and platform information to help triage the issue if
applicable:

* Waypoint CLI Version: 0.3.1
* Waypoint Server Platform and Version: any paltform, version 0.3.1 and older
* Waypoint Plugin: netlify

**Additional context**

### PR DESCRIPTION
**Pull request by izaaklauer**
Removing support for the netlify plugin. It is not currently used heavily enough to justify further development and maintenance.

Closes https://github.com/hashicorp/waypoint/issues/1468

### CODE REVIEW

**Review by evanphx — APPROVED** [review]
Looks clean to me! Saw your note about the UI, agree someone should chime in there first!

**Review by mitchellh — APPROVED** [review]

File: .changelog/1525.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:breaking-change
+plugin/netlify: Removing the netlify plugin
```
Comment (krantzinator) [inline .changelog/1525.txt:2]: **super nit/not blocking**: our changelog typically uses active verbs 
```suggestion
plugin/netlify: Removed the netlify plugin
```

**Review by krantzinator — APPROVED** [review]
🎉 🚀

**Review by briancain — APPROVED** [review]

**Review by jgwhite — APPROVED** [review]
Tried these changes out locally and all looks good to me 👍

**Pull request merged**